src/MySqlConnector/Protocol/Serialization/ByteBufferWriter.cs

```diff
@@ -226,7 +237,8 @@ private void Reallocate(int additional = 0)
 		m_output = new(m_buffer, usedLength, m_buffer.Length - usedLength);
 	}
 
-	private Encoder? m_encoder;
+	private static Encoder m_utf8Encoder = Encoding.UTF8.GetEncoder();
+	private static Encoder m_asciiEncoder = Encoding.ASCII.GetEncoder();
```

Comment:
Actually, `m_asciiEncoder` probably isn't necessary... the `Encoding.ASCII.GetBytes` static method is probably sufficient since we're encoding the whole string at once.

Reply:
`Encoding.ASCII.GetBytes` is also much faster than a cached `Encoder` object:

|      Method |                 Text |      Mean |     Error |    StdDev | Ratio | RatioSD |
|------------ |--------------------- |----------:|----------:|----------:|------:|--------:|
|  CachedUtf8 |                 1234 | 16.584 ns | 0.3541 ns | 0.4349 ns |  1.00 |    0.00 |
| CachedAscii |                 1234 | 14.951 ns | 0.1972 ns | 0.1845 ns |  0.91 |    0.03 |
|       Ascii |                 1234 |  5.975 ns | 0.1181 ns | 0.0986 ns |  0.36 |    0.01 |
|        Loop |                 1234 |  3.002 ns | 0.0853 ns | 0.0798 ns |  0.18 |    0.01 |
|             |                      |           |           |           |       |         |
|  CachedUtf8 | 1970-(...)00000 [26] | 19.331 ns | 0.4160 ns | 0.3891 ns |  1.00 |    0.00 |
| CachedAscii | 1970-(...)00000 [26] | 17.843 ns | 0.2783 ns | 0.2324 ns |  0.92 |    0.02 |
|       Ascii | 1970-(...)00000 [26] |  9.683 ns | 0.1879 ns | 0.1569 ns |  0.50 |    0.01 |
|        Loop | 1970-(...)00000 [26] | 20.023 ns | 0.4302 ns | 0.4955 ns |  1.03 |    0.03 |